pkg/apis/core/types.go

```diff
@@ -3575,30 +3575,33 @@ type ServiceSpec struct {
 	// More info: https://kubernetes.io/docs/concepts/services-networking/service/
 	Selector map[string]string
 
-	// ClusterIPs identifies all the ClusterIPs assigned to or requested for  this
-	// service. ClusterIPs are assigned or reserved  based on the values of
-	// Service.Spec.IPFamilies. Maximum of two entries (dualstack IPs) are
+	// ClusterIPs identifies all the ClusterIPs assigned to this
+	// service. ClusterIPs are assigned or reserved based on the values of
+	// service.spec.ipFamilies. A maximum of two entries (dual-stack IPs) are
 	// allowed in ClusterIPs. The IPFamily of each ClusterIP must match
-	// values provided in Service.Spec.IPFamilies
+	// values provided in service.spec.ipFamilies
 	// +optional
 	ClusterIPs []string
 
-	// IPFamilies identifies all the IPFamilies assigned to or requested for
-	// this Service. If value was not provided for IPFamilies it will be defaulted
-	// based on the cluster configuration and the value of Service.Spec.IPFamilyPolicy.
-	// Maximum of two values (dualstack IPFamilies) are allowed in IPFamilies.
+	// IPFamilies identifies all the IPFamilies assigned for this Service. If a value
+	// was not provided for IPFamilies it will be defaulted based on the cluster
+	// configuration and the value of service.spec.ipFamilyPolicy. A maximum of two
+	// values (dual-stack IPFamilies) are allowed in IPFamilies. IPFamilies field is
+	// conditionally immutable it allows for adding another IPFamily or removing
+	// the secondary IPFamily. But it does not allow changing the primary IPFamily
+	// of the service.
 	// +optional
 	IPFamilies []IPFamily
 
-	// IPFamilyPolicy represents the dualstack-ness requested or required by this
+	// IPFamilyPolicy represents the dual-stack-ness requested or required by this
 	// Service. If there is no value provided, then this Service will be considered
 	// SingleStack (single IPFamily). Services can be SingleStack (single IPFamily)
-	// PreferedDualStack (two dualstack IPFamilies on dualstack clusters) or (single
-	// IPFamily on singlestack clusters). Services can also be RequireDualStack where
-	// it will have two dualstack IPFamilies on dualstack configured clusters otherwise
-	// (one a single stack cluster) it will fail. The IPFamilies and ClusterIPs assigned
-	// to this service can be controlled by Service.Spec.IPFamilies and Service.Spec.ClusterIPs
-	// respectively
+	// PreferedDualStack (two dual-stack IPFamilies on dual-stack clusters or single
+	// IPFamily on single-stack clusters). Services can also be RequireDualStack where
```

Comment:
Weird sentence break. I suggest:

Services can be "SingleStack" (single IPFamily), "PreferDualStack" (two dual-stack IPFamilies on dual-stack clusters or single IPFamily on single-stack clusters), or "RequireDualStack" (two dual-stack IPFamilies on dual-stack configured clusters or error on single-stack clusters).